util/fipstools/test_fips.c

```diff
@@ -394,6 +396,46 @@ int main(int argc, char **argv) {
   ECDSA_SIG_free(sig);
   EC_KEY_free(ec_key);
 
+  /* Ed25519 */
+  printf("About to Ed25519 sign ");
+  hexdump(kPlaintextSHA256, sizeof(kPlaintextSHA256));
+  uint8_t ed_public_key[ED25519_PUBLIC_KEY_LEN];
+  uint8_t ed_private_key[ED25519_PRIVATE_KEY_LEN];
+  ED25519_keypair(ed_public_key, ed_private_key);
+  uint8_t ed_signature[ED25519_SIGNATURE_LEN];
+  if (!ED25519_sign(ed_signature,kPlaintextSHA256, sizeof(kPlaintextSHA256), ed_private_key) ||
+    !ED25519_verify(kPlaintextSHA256, sizeof(kPlaintextSHA256), ed_signature, ed_public_key)) {
+    printf("ED25519 Sign/Verify PWCT failed.\n");
```

Comment:
Is it only that the PWCT failed? Wouldn't the module have aborted if it was the case?

Reply:
Good point, test_fips.c only needs to use the functions to trigger the the lazy self tests. I was following the ECDSA pattern up above L385-394 but yes in theory this failure should not happen.